### PR TITLE
feat(apig): import api_gateway and apig (Dedicated api_gateway) resource…

### PR DESCRIPTION
**Pull request by Zhukun-Huawei**
**What this PR does / why we need it**:
add resource api_gateway_environment resource for flexibleengine
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
 Schema updated.

## Acceptance Steps Performed
```
api-gateway-environment
=== RUN   TestAccEnvironment_basic
=== PAUSE TestAccEnvironment_basic
=== CONT  TestAccEnvironment_basic
--- PASS: TestAccEnvironment_basic (162.50s)
PASS
coverage: 1.7% of statements in ../../../terraform-provider-flexibleengine/...
```
```
apig-acl-policy
=== RUN   TestAccAclPolicy_basic
=== PAUSE TestAccAclPolicy_basic
=== CONT  TestAccAclPolicy_basic
--- PASS: TestAccAclPolicy_basic (632.02s)
PASS
coverage: 2.1% of statements in ../../../terraform-provider-flexibleengine/...
```
```
apig-acl-policy-associate
=== RUN   TestAccAclPolicyAssociate_basic
=== PAUSE TestAccAclPolicyAssociate_basic
=== CONT  TestAccAclPolicyAssociate_basic
--- PASS: TestAccAclPolicyAssociate_basic (793.73s)
PASS
coverage: 3.8% of statements in ../../../terraform-provider-flexibleengine/...
```
```
apig-channel
=== RUN   TestAccChannel_basic
=== PAUSE TestAccChannel_basic
=== CONT  TestAccChannel_basic
--- PASS: TestAccChannel_basic (664.47s)
PASS
coverage: 4.0% of statements in ../../../terraform-provider-flexibleengine/...
```
```
apig-instance-routes
=== RUN   TestAccInstanceRoutes_basic
=== PAUSE TestAccInstanceRoutes_basic
=== CONT  TestAccInstanceRoutes_basic
--- PASS: TestAccInstanceRoutes_basic (544.33s)
PASS
coverage: 2.1% of statements in ../../../terraform-provider-flexibleengine/...
```

注：
image

api-acl-policy 这个资源的API文档无domain_id这个枚举值，单测也跑不通所以在本pr中去掉了这个枚举

### CODE REVIEW

**Comment by chengxiangdong** [comment]
pls resolve the conflicts

**Comment by Zhukun-Huawei** [comment]
> pls resolve the conflicts

done, i have rebased

File: docs/resources/api_gateway_environment.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+subcategory: "API Gateway (Shared APIG)"
```
Comment (chengxiangdong) [inline docs/resources/api_gateway_environment.md:2]:
```suggestion
subcategory: "API Gateway"
```

File: flexibleengine/acceptance/common.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ data "flexibleengine_compute_flavors_v2" "test" {
   memory_size       = 4
 }
 
-data "flexibleengine_images_image" "test" {
+data "flexibleengine_images_image_v2" "test" {
```
Comment (ShiChangkuo) [inline flexibleengine/acceptance/common.go:58]:
```suggestion
data "flexibleengine_images_image" "test" {
```

**Pull request merged**